tests/tests.cpp

```diff
@@ -50,6 +50,127 @@ TEST(Serialize64, serialize)
+    ASSERT_EQ(writestring.size(), checksize(sizeadded, sizeof(binary)));
+
+    w.serializecstr(cstr1.c_str(), true);
+    ASSERT_EQ(writestring.size(), checksize(sizeadded, 2 + cstr1.size() + 1));
+
+    w.serializecstr(cstr2.c_str(), false);
+    ASSERT_EQ(writestring.size(), checksize(sizeadded, 2 + cstr2.size()));
+
+    w.serializestring(stringtest);
+    ASSERT_EQ(writestring.size(), checksize(sizeadded, 2 + stringtest.size()));
+
+    w.serializei64(i64);
+    ASSERT_EQ(writestring.size(), checksize(sizeadded, 8));
+
+    w.serializeu32(u32);
+    ASSERT_EQ(writestring.size(), checksize(sizeadded, 4));
+
+    w.serializehandle(handle1);
+    ASSERT_EQ(writestring.size(), checksize(sizeadded, sizeof(handle)));
+
+    w.serializebool(b);
+    ASSERT_EQ(writestring.size(), checksize(sizeadded, sizeof(bool)));
+
+    w.serializebyte(by);
+    ASSERT_EQ(writestring.size(), checksize(sizeadded, 1));
+
+    w.serializechunkmacs(cm);
+    ASSERT_EQ(writestring.size(), checksize(sizeadded, 2 + 1 * (sizeof(m_off_t) + sizeof(ChunkMAC))));
+
+    w.serializeexpansionflags(1, 0, 1, 0, 0, 0, 1, 1);
+    ASSERT_EQ(writestring.size(), checksize(sizeadded, 8));
+
+    writestring += "abc";
+
+    // now read the serialized data back
+    string readstring = writestring;
+    CacheableReader r(readstring);
+
+    ::mega::byte check_binary[5];
+    string check_cstr1;
+    string check_cstr2;
+    string check_stringtest;
+    int64_t check_i64;
+    uint32_t check_u32;
+    handle check_handle1;
+    bool check_b;
+    ::mega::byte check_by;
+    chunkmac_map check_cm;
+
+    ASSERT_TRUE(r.unserializebinary(check_binary, sizeof(check_binary)));
+    ASSERT_EQ(0, memcmp(check_binary, binary, sizeof(binary)));
+
+    ASSERT_TRUE(r.unserializecstr(check_cstr1, true));
+    ASSERT_EQ(check_cstr1, cstr1);
+
+    ASSERT_TRUE(r.unserializecstr(check_cstr2, false));
+    ASSERT_EQ(check_cstr2, cstr2);
+
+    ASSERT_TRUE(r.unserializestring(check_stringtest));
+    ASSERT_EQ(check_stringtest, stringtest);
+
+    ASSERT_TRUE(r.unserializei64(check_i64));
+    ASSERT_EQ(check_i64, i64);
+
+    ASSERT_TRUE(r.unserializeu32(check_u32));
+    ASSERT_EQ(check_u32, u32);
+
+    ASSERT_TRUE(r.unserializehandle(check_handle1));
+    ASSERT_EQ(check_handle1, handle1);
+
+    ASSERT_TRUE(r.unserializebool(check_b));
+    ASSERT_EQ(check_b, b);
+
+    ASSERT_TRUE(r.unserializebyte(check_by));
+    ASSERT_EQ(check_by, by);
+
+    ASSERT_TRUE(r.unserializechunkmacs(check_cm));
+    ASSERT_EQ(check_cm[777].offset, cm[777].offset);
+
+    unsigned char expansions[8];
+    ASSERT_TRUE(r.unserializeexpansionflags(expansions));
+    ASSERT_EQ(expansions[0], 1);
+    ASSERT_EQ(expansions[1], 0);
+    ASSERT_EQ(expansions[2], 1);
+    ASSERT_EQ(expansions[3], 0);
+    ASSERT_EQ(expansions[4], 0);
+    ASSERT_EQ(expansions[5], 0);
+    ASSERT_EQ(expansions[6], 1);
+    ASSERT_EQ(expansions[7], 1);
+
+    r.eraseused(readstring);
+    ASSERT_EQ(readstring, "abc");
+}
+
+
```

Comment:
We might want to have some tests for serialize -> deserialize -> compare and unserialize -> serialize -> compare for different stored objects. But that can go out of the scope of this PR

Reply:
Yes - I've added one now for MediaProperties, and we can add more in the future when we use these in particular classes also. cheers